mewarpx/mewarpx/diags_store/particle_histogram_diagnostic.py

```diff
@@ -12,127 +12,91 @@
-            )
-            return
-        elif mwxrun.dim == 2:
-            self.linres_y = 1
-
-        if isinstance(self.assembly, assemblies.ZPlane):
-            self.linres_z = 1
-
-        super(ParticleHistDiag, self).__init__(diag_steps=diag_steps, **kwargs)
 
-        # for now we hard code this diagnostic to only save weight but this
-        # can be extended in the future
-        self.quantities = ['w']
+        super(BaseParticleHistDiag, self).__init__(
+            diag_steps=diag_steps, **kwargs)
 
         # install the scraper diag function
         self.assembly.add_diag_fn(
-            self._process_scraped_particles, ['x', 'y', 'z'] + self.quantities
+            self._process_scraped_particles, self.quantities
         )
 
         # create bins and data array after the simulation has been initialized
         # since the bin specifications are written to file
         callbacks.installafterinit(self.initialize)
 
-        # install callback function to save and reset histogram
-        callbacks.installafterstep(self.save_and_reset_histogram)
-
     def initialize(self):
         """Set up the histogram bins and data array."""
         self.setup_bins()
         self.setup_array()
 
     def setup_bins(self):
         """Calculate the bin edges."""
-        domain = [(mwxrun.xmin, mwxrun.xmax), (mwxrun.ymin, mwxrun.ymax),
-                  (mwxrun.zmin, mwxrun.zmax)]
-        linres = [self.linres_x+1, self.linres_y+1, self.linres_z+1]
+        # sanity check
+        if len(self.domain) != len(self.linres):
+            raise AttributeError(
+                "Number of domain ranges should match length of linres array."
+            )
 
         self.bins = []
-        for ii in range(3):
-            if linres[ii] <= 2:
+        for ii in range(len(self.domain)):
+            if self.linres[ii] <= 2:
                 self.bins.append([-np.inf, np.inf])
             else:
                 self.bins.append(
-                    list(np.linspace(domain[ii][0], domain[ii][1], linres[ii]))
+                    list(np.linspace(self.domain[ii][0], self.domain[ii][1],
+                         self.linres[ii]+1))
                 )
         self.bins = np.array(self.bins, dtype=object)
 
         # save bin details to file
-        fname = os.path.join(
-            self.write_dir, f"{self.assembly.name}_histogram_bins.npy"
-        )
-        np.save(fname, self.bins)
+        if mwxrun.me == 0:
+            fname = os.path.join(
+                self.write_dir, f"{self.assembly.name}_histogram_bins.npy"
+            )
+            np.save(fname, self.bins)
 
     def setup_array(self):
         """Calculate the necessary array size; create the array itself.
         """
-        shape = ((len(self.species_list),)
-                 + (self.linres_x, self.linres_y, self.linres_z,)
+        shape = ((len(self.species_list),) + tuple(self.linres)
                  + (len(self.quantities),))
```

Comment:
```suggestion
        shape = (len(self.species_list),) + tuple(self.linres)
```
`w` isn't its own dimension in the histogram, and other quantities in the future will be dimensions captured by the `linres` tuple; I don't think this is necessary.